orca/lockbox/programs/liquidity_lockbox/src/lib.rs

```diff
@@ -251,15 +265,23 @@ pub mod liquidity_lockbox {
     let position_liquidity = ctx.accounts.pda_lockbox_position.position_liquidity;
 
     // Check that the liquidity is not zero - must never happen if the total liquidity is not zero
-    if position_liquidity == 0 {
-      return Err(ErrorCode::LiquidityZero.into());
+    if position_liquidity < MIN_LIQUIDITY_AMOUNT {
+      return Err(ErrorCode::LiquidityTooLow.into());
     }
 
     // Check the requested amount to be smaller or equal than the position liquidity
     if amount > position_liquidity {
       return Err(ErrorCode::AmountExceedsPositionLiquidity.into());
     }
 
+    // Get the post-withdraw token remainder
+    let remainder: u64 = position_liquidity - amount;
+    // One needs to withdraw the full position on not try to lower the position remainder below the negligible level
+    // If the remainder is too low, select another lockbox position
+    if remainder > 0 && remainder < MIN_LIQUIDITY_AMOUNT {
+      return Err(ErrorCode::RemainderTooLow.into());
+    }
```

Comment:
what happen in the following scenarion:
1. A  deposits x and position has idx=1
2. B deposits x+ MIN_LIQUIDITY_AMOUNT-1  and position has idx=2
3. B withdraws x from position with idx=1.
There is no way for A to recover x back. 
Clearly there is no way for B as well to recover MIN_LIQUIDITY_AMOUNT-1 , but is MIN_LIQUIDITY_AMOUNT is quite small this can be an issue

Reply:
This has been discussed.